mmhuman3d/core/cameras/camera_parameters.py

```diff
@@ -483,12 +487,45 @@ def __check_value_type__(self, key: Any, val: Any) -> None:
         Raises:
             TypeError:
                 Value is supported but doesn't match definition.
+
+        Returns:
+            val (Any): The value casted into correct format.
         """
-        if type(val) != self.__class__.SUPPORTED_KEYS[key]['type']:
+        np_type_mapping = {int: np.integer, float: np.floating}
+        supported_keys = self.__class__.SUPPORTED_KEYS
+        check_passed = False
+        ret_val = None
+        if supported_keys[key]['type'] == int or\
+                supported_keys[key]['type'] == float:
+            type_str = str(type(val))
+            if type(val) == self.__class__.SUPPORTED_KEYS[key]['type']:
+                check_passed = True
+                ret_val = val
+            if 'numpy' in type_str:
```

Comment:
There is a situation that `check_passed` may be set twice. Currently it seems will not happen cause all types in `SUPPORTED_KEYS` is python naive type. But it may happen when someone tries to add a new key, e.g.,
```
SUPPORTED_KEYS = {
    'X': np.int8,
}
```

Reply:
Yes. An `elif` is expected in the following line, will be fixed soon.

Reply:
Fixed in c3edd34.